examples/air-purifier-app/air-purifier-common/src/air-purifier-manager.cpp

```diff
@@ -209,60 +273,61 @@ void AirPurifierManager::PercentSettingWriteCallback(uint8_t aNewPercentSetting)
 
 void AirPurifierManager::SpeedSettingWriteCallback(uint8_t aNewSpeedSetting)
 {
-    if (aNewSpeedSetting != speedCurrent)
+    ChipLogDetail(NotSpecified, "AirPurifierManager::SpeedSettingWriteCallback: %d", aNewSpeedSetting);
+    if (mOnOffClusterOn)
     {
-        ChipLogDetail(NotSpecified, "AirPurifierManager::SpeedSettingWriteCallback: %d", aNewSpeedSetting);
-        speedCurrent  = aNewSpeedSetting;
-        Status status = FanControl::Attributes::SpeedCurrent::Set(mEndpointId, speedCurrent);
+        Status status = FanControl::Attributes::SpeedCurrent::Set(mEndpointId, aNewSpeedSetting);
         if (status != Status::Success)
         {
             ChipLogError(NotSpecified, "AirPurifierManager::SpeedSettingWriteCallback: failed to set SpeedCurrent attribute: %d",
                          to_underlying(status));
         }
+    }
 
-        // Determine if the speed change should also change the fan mode
-        if (speedCurrent == 0)
-        {
-            FanControl::Attributes::FanMode::Set(mEndpointId, FanControl::FanModeEnum::kOff);
-        }
-        else if (speedCurrent <= FAN_MODE_LOW_UPPER_BOUND)
-        {
-            FanControl::Attributes::FanMode::Set(mEndpointId, FanControl::FanModeEnum::kLow);
-        }
-        else if (speedCurrent <= FAN_MODE_MEDIUM_UPPER_BOUND)
-        {
-            FanControl::Attributes::FanMode::Set(mEndpointId, FanControl::FanModeEnum::kMedium);
-        }
-        else if (speedCurrent <= FAN_MODE_HIGH_UPPER_BOUND)
-        {
-            FanControl::Attributes::FanMode::Set(mEndpointId, FanControl::FanModeEnum::kHigh);
-        }
+    // Determine if the speed change should also change the fan mode
+    if (aNewSpeedSetting == 0)
+    {
+        FanControl::Attributes::FanMode::Set(mEndpointId, FanControl::FanModeEnum::kOff, MarkAttributeDirty::kIfChanged);
```

Comment:
Again, could just leave off the last arg here and various places below.  But either way.